source/extensions/stat_sinks/hystrix/hystrix.cc

```diff
@@ -0,0 +1,371 @@
+#include "extensions/stat_sinks/hystrix/hystrix.h"
+
+#include <chrono>
+#include <ctime>
+#include <iostream>
+#include <sstream>
+
+#include "common/buffer/buffer_impl.h"
+#include "common/common/logger.h"
+
+#include "server/http/admin.h"
+
+#include "absl/strings/str_cat.h"
+
+namespace Envoy {
+namespace Extensions {
+namespace StatSinks {
+
+const uint64_t HystrixStatCache::DEFAULT_NUM_OF_BUCKETS;
+
+// Add new value to rolling window, in place of oldest one.
+void HystrixStatCache::pushNewValue(const std::string& key, uint64_t value) {
+  // Create vector if do not exist.
+  // TODO trabetti: why resize + value param didn't work without the if?
+  if (rolling_stats_map_.find(key) == rolling_stats_map_.end()) {
+    rolling_stats_map_[key].resize(window_size_, value);
+  } else {
+    rolling_stats_map_[key][current_index_] = value;
+  }
+}
+
+uint64_t HystrixStatCache::getRollingValue(absl::string_view cluster_name, absl::string_view stat) {
```

Comment:
Can you change this method to return a structure of all the relevant stats for a cluster?  At the call-sites I only see static strings passed in for 'stat'.